### PR TITLE
initial xml table support

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -582,12 +582,37 @@ class Parser(metaclass=_Parser):
     }
 
     CONSTRAINT_PARSERS = {
-        TokenType.CHECK: lambda self: self.expression(
-            exp.Check, this=self._parse_wrapped(self._parse_conjunction)
+        "AUTOINCREMENT": lambda self: self._parse_auto_increment(),
```
Comment (tobymao) [inline sqlglot/parser.py:585]: all of these can be changed to : self._parse_auto_increment

File: sqlglot/parser.py [hunk 2]
```diff
@@ -3010,6 +3015,13 @@ def _parse_foreign_key(self) -> exp.Expression:
         )
 
     def _parse_primary_key(self) -> exp.Expression:
+        desc = None
```
Comment (tobymao) [inline sqlglot/parser.py:3018]: desc = (self.Match(ASC) or ...) and token_type == TokenType.desc

File: sqlglot/parser.py [hunk 3]
```diff
@@ -582,12 +582,37 @@ class Parser(metaclass=_Parser):
     }
 
     CONSTRAINT_PARSERS = {
-        TokenType.CHECK: lambda self: self.expression(
```
Comment (tobymao) [inline sqlglot/parser.py:585]: can you delete all t he tokens? that are no longer used

File: sqlglot/parser.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlglot/parser.py:585]: We need the lambdas because we can't reference `self` there, right?

File: sqlglot/parser.py (hunk 2 above)
Comment (GeorgeSittas) [reply sqlglot/parser.py]: Rewrote it with `self._match_set` instead.

File: sqlglot/parser.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/parser.py:585]: ah right

File: sqlglot/parser.py (hunk 3 above)
Comment (GeorgeSittas) [reply sqlglot/parser.py:585]: got rid of some of them

**Comment by GeorgeSittas** [comment]
Ok should be good

**Pull request merged**